fedimint-core/src/api.rs

```diff
@@ -511,14 +517,28 @@ struct GlobalFederationApiWithCache<T> {
     /// synchronize, or split into a handful of groups. And if they are not,
     /// no LRU here is going to help them.
     #[allow(clippy::type_complexity)]
-    await_block_lru: Arc<tokio::sync::Mutex<lru::LruCache<u64, Arc<OnceCell<SessionOutcome>>>>>,
+    await_session_lru: Arc<tokio::sync::Mutex<lru::LruCache<u64, Arc<OnceCell<SessionOutcome>>>>>,
```

Comment:
This cache hasn't been released yet, right? Could we just remove it now that it's unused? We could also wait for the deprecation of the `await_block` API.

Reply:
Oh, are we planning to remove `await_block`? Seemed like a useful API to have as it's safer to use (can't ever mishandle partially filed session) and the only way to actually cleanly block waiting for session to close without retrying. @elsirion

Reply:
I mean, do we still use it anywhere after this PR?

I'd be ok with keeping it if you only wanted to fetch the last session via the new API, in which case that wouldn't really need caching unless we expect to have 100s of modules running in a client.

But if we fully switch to the new API as done in this PR afaik, then there is no need to keep the old one around.

Reply:
Just today we were writing tests that require waiting for a session to close, so it seems useful as a general purpose API. @justinmoon was writing a test checking if with one peer offline sessions close.

Reply:
Anyway, I think we should land, and decide to delete (or not) this endpoint elsewhere.

This cache is probably like 64B until something actually uses it, so no point in increasing the scope of the change anymore.